sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpAsyncResponse.java

```diff
@@ -108,14 +109,19 @@ public Mono<InputStream> getBodyAsInputStream() {
     @Override
     public void writeBodyTo(WritableByteChannel channel) throws IOException {
         if (responseBody != null) {
-            IOUtils.transfer(responseBody.source(), channel);
+            try {
+                IOUtils.transfer(responseBody.source(), channel, responseBody.contentLength());
+            } finally {
+                close();
+            }
         }
     }
 
     @Override
     public Mono<Void> writeBodyToAsync(AsynchronousByteChannel channel) {
         if (responseBody != null) {
-            return IOUtils.transferAsync(responseBody.source(), channel);
+            return IOUtils.transferAsync(responseBody.source(), channel, responseBody.contentLength())
+                .doFinally(ignored -> close());
```

Comment:
Not blocking the pr but just sharing, the `doFinally` runs in the reverse order of reactor chain assembling that application formed, so it could cause the `close` call to delay more than it needs to be.  An alternate `Mono.UsingWhen` or `.doOnEach(voidSignal -> { close(); })` could be used which can immediately `close` upon termination of transfer. Something for us to consider in the future after necessary evaluation.

Reply:
I'll update this in the next commit